libkbfs/block_retrieval_queue.go

```diff
@@ -218,6 +241,17 @@ func (brq *blockRetrievalQueue) FinalizeRequest(retrieval *blockRetrieval, block
 	// lock.
 	retrieval.reqMtx.Lock()
 	defer retrieval.reqMtx.Unlock()
+	// Cache the block if there is no error.
+	if brq.cacheFunc() != nil && err == nil {
+		_ = brq.cacheFunc().Put(retrieval.blockPtr, retrieval.kmd.TlfID(), block, retrieval.cacheLifetime)
+	}
+	go func() {
```

Comment:
Any reason why this isn't just done before `reqMtx.Lock()`, perhaps guarded with `err == nil`? the prefetcher methods all return immediately, right?